### ISSUE TITLE
The active test run was aborted. Reason: Unable to communicate with test host process.

### ISSUE THREAD

**Issue by EnriGrossi** [microsoft/azure-pipelines-tasks#11818]
**Question, Bug, or Feature?**  
*Type*: Question

**Enter Task Name**: Visual Studio Test (VsTest) 

## Environment

- Server - TFS on-premises 
- Agent - Private (agent version: '2.153.9')

## Issue Description

Running test step is aborted with this message: `The active test run was aborted. Reason: Unable to communicate with test host process`

I have no further information about what this failure depends on and I have already tried, unsuccessfully, to activate DisableAppDomain.

Any suggestion?

### Task logs

[stripedlog.txt](https://github.com/microsoft/azure-pipelines-tasks/files/3873983/stripedlog.txt)

**Comment by ShreyasRmsft** [comment]
@EnriGrossi how often is this happening? 

In your runsettings please set the

`<DisableAppDomain>` node value to true.

Links for help on why and how to do this:

https://github.com/microsoft/vstest-docs/blob/master/RFCs/0019-Disable-Appdomain-While-Running-Tests.md
https://github.com/microsoft/vstest-docs/blob/master/docs/configure.md

**Comment by EnriGrossi** [comment]
@ShreyasRmsft the problem is repetitive.

I've already tried to use `<DisableAppDomain>` (you can also see this tag in the log)

Thank you very much.
Enri

**Comment by ShreyasRmsft** [comment]
Could change the advanced diagnostics option to collect dump Always and see if you can get a memory dump when the crash occurs?

From the logs it seems like the testhost is crashing during assembly cleanup or during appdomain tear down. This generally happens when the tests leave a handle to some resource open or some thread is still running.

**Comment by EnriGrossi** [comment]
I've already enabled diagnostic log to connect dump:

![image](https://user-images.githubusercontent.com/12607857/69410808-14ebfc80-0d0c-11ea-8364-a13a541b9e83.png)

but no dump file is created in the set folder

**Comment by ShreyasRmsft** [comment]
Can you please enabled the option in the task. Right now from the logs it is set to collect on abort only. Set it to always 

![image](https://user-images.githubusercontent.com/25194518/69412796-973ce680-0d35-11ea-9c3e-30e0bfce6d66.png)

**Comment by EnriGrossi** [comment]
Here the dump file:


[testhost_8076_65a00340dc484ce2a63e02724f4d281a.zip](https://github.com/microsoft/azure-pipelines-tasks/files/3878988/testhost_8076_65a00340dc484ce2a63e02724f4d281a.zip)

Thank you again
Enri

**Comment by ShreyasRmsft** [comment]
I'm unable to see anything that i could read in the dump, with the pdbs on your machine if any of your test code was still running you should be able to see some thread with code you might recognize, then again it could be some native handle.

**Comment by ShreyasRmsft** [comment]
Would it be possible for you to share a sample repro project with us where we could reproduce this locally? This issue seems very specific to your test code.

**Comment by ShreyasRmsft** [comment]
@EnriGrossi any updates here?

**Comment by EnriGrossi** [comment]
Hi @ShreyasRmsft, I understood which test causes the problem and at the moment I disabled it.
This problem began to manifest itself when we switched from the 2015 visual studio tools to the current ones (2019).
I'm still trying to figure out how to reproduce the problem (at the moment it is difficult for me to pass the code to you because some of our libraries are used in the test).

Thank you very much
Enri
